packages/plugin/src/browser/hosted-plugin-manager-client.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from 'inversify';
+import URI from '@theia/core/lib/common/uri';
+import { MessageService } from '@theia/core/lib/common';
+import { LabelProvider, isNative } from '@theia/core/lib/browser';
+import { WindowService } from '@theia/core/lib/browser/window/window-service';
+import { WorkspaceService } from '@theia/workspace/lib/browser';
+import { FileSystem } from '@theia/filesystem/lib/common';
+import { FileDialogFactory, DirNode } from '@theia/filesystem/lib/browser';
+import { HostedPluginServer } from '../common/plugin-protocol';
+import { HostedPluginCommands } from './plugin-api-frontend-contribution';
+
+@injectable()
+export class HostedPluginClient {
+    @inject(HostedPluginServer)
+    protected readonly hostedPluginServer: HostedPluginServer;
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+    @inject(FileDialogFactory)
+    protected readonly fileDialogFactory: FileDialogFactory;
+    @inject(LabelProvider)
+    protected readonly labelProvider: LabelProvider;
+    @inject(WindowService)
+    protected readonly windowService: WindowService;
+    @inject(FileSystem)
+    protected readonly fileSystem: FileSystem;
+    @inject(WorkspaceService)
+    protected readonly workspaceService: WorkspaceService;
+
+    protected pluginLocation: URI | undefined;
+    protected pluginInstanceUri: string | undefined;
+
+    run() {
```

Comment:
should it be start()/stop() instead of run()/terminate() ? it doesn't seem to be a thread

Reply:
Ok, I'll rename them